src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // Wraps a record
+    // This also preserves the lazy field enumeration semantics of a recordType.
+    internal class SymbolTableOverRecordType : ReadOnlySymbolTable, IGlobalSymbolNameResolver
+    {
+        private readonly RecordType _type;
+
+        public SymbolTableOverRecordType(RecordType type)
+        {
+            _type = type;
+            _debugName = "per-eval";
+        }
+
+        IReadOnlyDictionary<string, NameLookupInfo> IGlobalSymbolNameResolver.GlobalSymbols
+        {
+            get
+            {
+                // $$$ Enumeration is for intellisense, should return in DisplayNames?
+                var map = new Dictionary<string, NameLookupInfo>();
+                foreach (var kv in _type.GetFieldTypes())
+                {
+                    map[kv.Name] = Create(kv.Name, kv.Type);
+                }
+
+                return map;
+            }
+        }
+
+        internal override bool TryLookup(DName name, out NameLookupInfo nameInfo)
+        {
+            // $$$ Not handling DisplayNames here.... 
+            // Dtypoe.TryGetDisplayNameForColumn? 
```

Comment:
Should be the same as other symbol table lookups, like what was previously in PfxConfig, but pulling the displaynameprovider from the _type
```
            if (_type._type.DisplayNameProvider.TryGetDisplayName(name, out displayName))
            {
                lookupName = name;
            }
            else if (_type._type.DisplayNameProvider.TryGetLogicalName(name, out var logicalName))
            {
                lookupName = logicalName;
                displayName = name;
            }

            __type.TryGetFieldType(lookupName, out var type)
```
and ensure the DisplayName is set in the NameLookupInfo that's returned

Reply:
This is a little tricky if all the symbol tables need to handle it themselves. Maybe we add `DisplayNameProvider` to the SymbolTable interface, and handle it in a wrapper?

Reply:
discussed offlne. 
We do need a display name lookup here.  I'll do what rowscope does.